make triton kernel usage user controlable

Gasoonjia opened the pull request:
Summary: We observed perf regression on some models after using triton sdpa kernel, compare with decomposed operator. This diff makes triton kernel controalable by user to prevent such perf regression.

Reviewed By: larryliu0820

Differential Revision: D88096054

larryliu0820 reviewed (approved):
Review automatically exported from Phabricator review in Meta.

Gasoonjia committed:
make triton kernel usage user controlable (#16030)

Summary:

We observed perf regression on some models after using triton sdpa kernel, compare with decomposed operator. This diff makes triton kernel controalable by user to prevent such perf regression.

Reviewed By: larryliu0820

Differential Revision: D88096054
